src/scanner.c

```diff
@@ -187,10 +195,35 @@ ts_scanner_start_scan(ScannerCtx *ctx, InternalScannerCtx *ictx)
 
 	ictx->sctx = ctx;
 	ictx->closed = false;
+	ictx->registered_snapshot = false;
 
 	scanner = scanner_ctx_get_scanner(ctx);
 
 	scanner->openscan(ictx);
+
+	if (ctx->snapshot == NULL)
+	{
+		/* Use SnapshotSelf by default. Since Scanner is mostly used for
+		 * internal metadata (chunks, slices, etc.), a scan should see every
+		 * change that has happened prior to the scan, including any changes
+		 * made by the scan itself.
+		 *
+		 * This is important in case of, e.g., concurrent chunk creation by
+		 * two transactions; we'd like transactions to see the new chunk as
+		 * soon as possible so that they don't try to create the same chunk,
+		 * leading to a conflict.
+		 *
+		 * Note, however, that this slightly violates SERIALIZED mode for
```

Comment:
I am confused by this comment. If we have 2 transactions creating metadata, we want to see only the "committed" data.  If I understand the SnapshotSelf semantics and your comments correctly :  it suggests that dirty reads are ok. Say, we are creating a dimension slice in 2 different txns, 2nd one sees an already-created slice and doesn't create it. Txn 1 aborts. Txn2 has bad metadata. Could you elaborate on what this change is and why it is required?

Reply:
The PR doesn't change the current behavior; we're still using `SnapshotSelf`. The change introduces the ability to use another snapshot when scanning data.  `SnapshotSelf` won't show "dirty" reads (i.e., data from other ongoing transactions), but it is an "instant" snapshot and shows changes made by the current command.

The comment is only to elaborate on existing behavior, not new behavior. The fact is, however, that using `SnapshotSelf` requires explanation because it isn't really a common snapshot to use and doesn't have proper MVCC semantics. But changing it to a MVCC snapshot (which is something we might consider) changes our behavior in SERIALIZABLE mode. 

For instance, if we use a MVCC snapshot using, e.g., `GetLatestSnapshot()` the `serializable_insert` test will fail with `could not serialize access due to read/write dependencies among transactions`. This is because an insert will create chunk metadata in one transaction and another concurrent transaction is not allowed to read that data in SERIALIZABLE mode as data read/writes must be serialized. This makes sense for user data, but probably not for metadata, or else the behavior of hypertables in the `serializable_insert` test will be different from regular tables (since they aren't creating chunks in transactions).

Reply:
FWIW, separate issues here: https://github.com/timescale/timescaledb/issues/2169